src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,144 @@
+        reader.read(buffer, kAtomHeaderSize);
+        byteRead += boxSize;
+        if (ATOM_TYPE(buffer + 4) == path[pathIdx]) {
+            return seekTillAtom(reader, path, ATOM_TYPE(buffer) - kAtomHeaderSize, pathIdx + 1);
+        }
+        reader.skip(ATOM_TYPE(buffer) - kAtomHeaderSize);
+    }
+    return -1;
+}
+
+} // anonymous namespace
+
+// static
+bool StemInfoImporter::isStemFile(
+        const QString& fileName) {
+    for (const QString& ext : kStemFileExtensions) {
+        if (fileName.endsWith(ext)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+QList<StemInfo> StemInfoImporter::importStemInfos(
+        const QString& filePath) {
+    // Fetch the STEM manifest which contain stream details
+    auto file = QFile(filePath);
+    if (!file.open(QIODeviceBase::ReadOnly)) {
+        kLogger.warning()
+                << "Failed to open input file"
+                << filePath;
+        return {};
+    }
+
+    uint32_t manifestSize;
+    if (!(manifestSize = seekTillAtom(file, kStemManifestAtomPath))) {
+        kLogger.warning()
+                << "Failed to find the steam manifest in the file"
+                << filePath;
+        return {};
+    }
+
+    auto jsonData = QJsonDocument::fromJson(file.read(manifestSize));
+    VERIFY_OR_DEBUG_ASSERT(jsonData.isObject()) {
+        kLogger.warning()
+                << "Failed to extract the manifest data"
+                << filePath;
+        return {};
+    };
+
+    auto json = jsonData.object();
+
+    // Check the manifest version
+    auto version = json.value("version").toInteger(-1);
+    if (version <= 0) {
+        kLogger.debug() << "Unexpected or missing version value in STEM manifest";
+        return {};
+    } else if (version > kSupportedStemVersion) {
+        kLogger.debug() << "Unsupported stem version" << version << "but trying anyway";
+    }
+
+    // Extract stem metadata
+    auto stems = json.value("stems");
+    if (!stems.isArray()) {
+        kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+        return {};
+    }
+    auto stemArray = stems.toArray();
+    QList<StemInfo> stemsList;
+    stemsList.reserve(stemArray.size());
+    for (const auto& stemRef : stemArray) {
+        if (!stemRef.isObject()) {
+            kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+            return {};
+        }
+        auto stem = stemRef.toObject();
+        auto color = QColor(stem.value("color").toString());
+        auto name = stem.value("name").toString();
+        if (!color.isValid() || name.isEmpty()) {
+            kLogger.debug() << "Unexpected or missing stem name or attribute in STEM manifest";
+            return {};
+        }
+        stemsList.emplace_back(name, color);
+    }
+
+    // Extract DSP information
+    // TODO(XXX) DSP only contains a limiter and a compressor effect, which
+    // Mixxx doesn't have yet. Parse and implement when supported
```

Comment:
nit: maybe worth printing a warning if we detected enabled compressor/ limiter the fact that it will be ignored.

Reply:
Do you mean a modal popup? If you mean log messages, I'm not sure what would be the value of it, but happy to put a debug a message to reduce noise.

Reply:
This information would not be helpful for a normal user. A warning in the log should be enough.

Reply:
I'll put a debug message